server/src/test/java/org/elasticsearch/index/engine/ThreadPoolMergeSchedulerTests.java

```diff
@@ -535,10 +536,10 @@ public void testSchedulerCloseWaitsForRunningMerge() throws Exception {
                         threadPoolMergeScheduler.merge(mergeSource2, randomFrom(MergeTrigger.values()));
                         // when the merge scheduler is closed it won't pull in any new merges from the merge source
                         verifyNoInteractions(mergeSource2);
+                        // assert the merge still shows up as "running"
+                        assertThat(threadPoolMergeScheduler.getRunningMergeTasks().keySet(), contains(oneMerge));
```

Comment:
I wonder if we should rather just wait for the merge thread to have picked up the task - before this assert busy block? Using `CyclicBarrier` for instance. Then I think the original check here can work too - without the waiting for it using assert busy. I think that reads more clearly and is a bit stronger.

Reply:
Pushed [4131809](https://github.com/elastic/elasticsearch/pull/126110/commits/413180993b1b054032016010e8fa9a0c8103330a)